From ea9b8497e3d5b8323761801d59caae856634e6fc Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Thu, 22 Sep 2022 10:49:20 +0530
Subject: [PATCH] ath12k : enhance peer create/delete log

print the peer count during peer create/delete
and during vdev delete.

This can help when we face issues related to peer count
and peer delete/vdev delete sequence.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c  | 25 ++++++++++++++-----------
 drivers/net/wireless/ath/ath12k/peer.c |  5 +++--
 drivers/net/wireless/ath/ath12k/wmi.c  | 18 +++++++++++-------
 3 files changed, 28 insertions(+), 20 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 0324451..3f1b36e 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -1034,6 +1034,8 @@ void ath12k_mac_peer_cleanup_all(struct ath12k *ar)
 
 	ar->num_peers = 0;
 	ar->num_stations = 0;
+
+	ath12k_dbg(ar->ab, ATH12K_DBG_MAC, "ath12k mac peer cleanup done\n");
 }
 
 static int ath12k_mac_vdev_setup_sync(struct ath12k *ar)
@@ -4082,7 +4084,8 @@ static int ath12k_mac_vdev_delete(struct ath12k *ar, struct ieee80211_vif *vif)
 	time_left = wait_for_completion_timeout(&ar->vdev_delete_done,
 						ATH12K_VDEV_DELETE_TIMEOUT_HZ);
 	if (time_left == 0) {
-		ath12k_warn(ar->ab, "Timeout in receiving vdev delete response\n");
+		ath12k_warn(ar->ab, "Timeout in receiving vdev delete response vdev_id : %d\n",
+			    arvif->vdev_id);
 		ret = -ETIMEDOUT;
 		goto clean_up;
 	}
@@ -5414,8 +5417,8 @@ static int ath12k_mac_station_add(struct ath12k *ar,
 		goto free_rx_stats;
 	}
 
-	ath12k_dbg(ab, ATH12K_DBG_MAC, "Added peer: %pM for VDEV: %d\n",
-		   sta->addr, arvif->vdev_id);
+	ath12k_dbg(ab, ATH12K_DBG_MAC, "Added peer: %pM for VDEV: %d num_stations : %d\n",
+		   sta->addr, arvif->vdev_id, ar->num_stations);
 
 	if (ath12k_debugfs_is_extd_tx_stats_enabled(ar) && (!arsta->tx_stats)) {
 		arsta->tx_stats = kzalloc(sizeof(*arsta->tx_stats),
@@ -5557,18 +5560,18 @@ static int ath12k_mac_op_sta_state(struct ieee80211_hw *hw,
 
 		ret = ath12k_peer_delete(ar, arvif->vdev_id, sta->addr);
 		if (ret)
-			ath12k_warn(ar->ab, "Failed to delete peer: %pM for VDEV: %d\n",
-				    sta->addr, arvif->vdev_id);
+			ath12k_warn(ar->ab, "Failed to delete peer: %pM for VDEV: %d num_peers : %d\n",
+				    sta->addr, arvif->vdev_id, ar->num_peers);
 		else
-			ath12k_dbg(ar->ab, ATH12K_DBG_MAC, "Removed peer: %pM for VDEV: %d\n",
-				   sta->addr, arvif->vdev_id);
+			ath12k_dbg(ar->ab, ATH12K_DBG_MAC, "Removed peer: %pM for VDEV: %d num_peers : %d\n",
+				   sta->addr, arvif->vdev_id, ar->num_peers);
 
 		ath12k_mac_dec_num_stations(arvif, sta);
 		spin_lock_bh(&ar->ab->base_lock);
 		peer = ath12k_peer_find(ar->ab, arvif->vdev_id, sta->addr);
 		if (peer && peer->sta == sta) {
-			ath12k_warn(ar->ab, "Found peer entry %pM n vdev %i after it was supposedly removed\n",
-				    vif->addr, arvif->vdev_id);
+			ath12k_warn(ar->ab, "Found peer entry %pM n vdev %d after it was supposedly removed"
+				    " num_peers : %d\n", vif->addr, arvif->vdev_id, ar->num_peers);
 			peer->sta = NULL;
 			list_del(&peer->list);
 			kfree(peer);
@@ -8075,8 +8078,8 @@ static void ath12k_mac_op_remove_interface(struct ieee80211_hw *hw,
 	if (arvif->vdev_type == WMI_VDEV_TYPE_AP) {
 		ret = ath12k_peer_delete(ar, arvif->vdev_id, vif->addr);
 		if (ret)
-			ath12k_warn(ab, "failed to submit AP self-peer removal on vdev %d: %d\n",
-				    arvif->vdev_id, ret);
+			ath12k_warn(ab, "failed to submit AP self-peer removal on vdev %d: %d"
+				    " num_peer : %d\n", arvif->vdev_id, ret, ar->num_peers);
 	}
 
 	ret = ath12k_mac_vdev_delete(ar, vif);
diff --git a/drivers/net/wireless/ath/ath12k/peer.c b/drivers/net/wireless/ath/ath12k/peer.c
index dae3fee..c2ec1b8 100644
--- a/drivers/net/wireless/ath/ath12k/peer.c
+++ b/drivers/net/wireless/ath/ath12k/peer.c
@@ -217,14 +217,15 @@ int ath12k_wait_for_peer_delete_done(struct ath12k *ar, u32 vdev_id,
 
 	ret = ath12k_wait_for_peer_deleted(ar, vdev_id, addr);
 	if (ret) {
-		ath12k_warn(ar->ab, "failed wait for peer deleted");
+		ath12k_warn(ar->ab, "failed wait for peer deleted peer_addr : %pM\n", addr);
 		return ret;
 	}
 
 	time_left = wait_for_completion_timeout(&ar->peer_delete_done,
 						3 * HZ);
 	if (time_left == 0) {
-		ath12k_warn(ar->ab, "Timeout in receiving peer delete response\n");
+		ath12k_warn(ar->ab, "Timeout in receiving peer delete response peer_addr : %pM\n",
+			    addr);
 		return -ETIMEDOUT;
 	}
 
diff --git a/drivers/net/wireless/ath/ath12k/wmi.c b/drivers/net/wireless/ath/ath12k/wmi.c
index a37c78c..038ae2b 100644
--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -1087,7 +1087,8 @@ int ath12k_wmi_vdev_delete(struct ath12k *ar, u8 vdev_id)
 						 sizeof(*cmd));
 	cmd->vdev_id = cpu_to_le32(vdev_id);
 
-	ath12k_dbg(ar->ab, ATH12K_DBG_WMI, "WMI vdev delete id %d\n", vdev_id);
+	ath12k_dbg(ar->ab, ATH12K_DBG_WMI, "WMI vdev delete id %d num_peers : %d\n",
+		   vdev_id, ar->num_peers);
 
 	ret = ath12k_wmi_cmd_send(wmi, skb, WMI_VDEV_DELETE_CMDID);
 	if (ret) {
@@ -1147,7 +1148,8 @@ int ath12k_wmi_vdev_down(struct ath12k *ar, u8 vdev_id)
 
 	ret = ath12k_wmi_cmd_send(wmi, skb, WMI_VDEV_DOWN_CMDID);
 	if (ret) {
-		ath12k_warn(ar->ab, "failed to submit WMI_VDEV_DOWN cmd\n");
+		ath12k_warn(ar->ab, "failed to submit WMI_VDEV_DOWN cmd vdev id : %d\n",
+			    vdev_id);
 		dev_kfree_skb(skb);
 	}
 
@@ -1348,8 +1350,8 @@ int ath12k_wmi_send_peer_create_cmd(struct ath12k *ar,
 	cmd->vdev_id = cpu_to_le32(arg->vdev_id);
 
 	ath12k_dbg(ar->ab, ATH12K_DBG_WMI,
-		   "WMI peer create vdev_id %d peer_addr %pM\n",
-		   arg->vdev_id, arg->peer_addr);
+				"WMI peer create vdev_id %d peer_addr %pM num_peer : %d\n",
+				param->vdev_id, param->peer_addr, ar->num_peers);
 
 	ret = ath12k_wmi_cmd_send(wmi, skb, WMI_PEER_CREATE_CMDID);
 	if (ret) {
@@ -1380,12 +1382,14 @@ int ath12k_wmi_send_peer_delete_cmd(struct ath12k *ar,
 	cmd->vdev_id = cpu_to_le32(vdev_id);
 
 	ath12k_dbg(ar->ab, ATH12K_DBG_WMI,
-		   "WMI peer delete vdev_id %d peer_addr %pM\n",
-		   vdev_id,  peer_addr);
+		   "WMI peer delete vdev_id %d peer_addr %pM num_peer : %d\n",
+		   vdev_id,  peer_addr, ar->num_peers);
 
 	ret = ath12k_wmi_cmd_send(wmi, skb, WMI_PEER_DELETE_CMDID);
 	if (ret) {
-		ath12k_warn(ar->ab, "failed to send WMI_PEER_DELETE cmd\n");
+		ath12k_warn(ar->ab, "failed to send WMI_PEER_DELETE cmd"
+			   " peer_addr %pM num_peer : %d\n",
+			    peer_addr, ar->num_peers);
 		dev_kfree_skb(skb);
 	}
 
-- 
2.17.1

